### PR TITLE
Experiment with adding drag&drop to image galleries

### PR DESCRIPTION
**Pull request by oandregal**
Related https://github.com/WordPress/gutenberg/issues/743

This PR experiments with adding Drag&Drop support for moving images within a gallery.

Peek 2019-04-26 19-27

### CODE REVIEW

**Comment by hacknug** [comment]
Looks good! Should we also add some kind of visual clue that let the users know what the order would be before `mouseup`?

**Comment by nosolosw** [comment]
> Should we also add some kind of visual clue that let the users know what the order would be before mouseup?

Yeah, design-wise needs improvement. Note that at the moment it uses the same mechanism (blue line indicating where it'll be dropped). I've got a working version of live DnD (you see where a block/image will be dropped during the dragging process). Perhaps it makes sense to land that first.

File: packages/block-editor/src/components/block-list/block.js [hunk 1]
```diff
@@ -248,7 +248,10 @@ export class BlockListBlock extends Component {
 	 * @return {void}
 	 */
 	preventDrag( event ) {
-		event.preventDefault();
+		const { type } = JSON.parse( event.dataTransfer.getData( 'text' ) );
+		if ( type !== 'image' ) {
```
Comment (simison) [inline packages/block-editor/src/components/block-list/block.js]: Does this create an exception to `preventDrag` for all image elements within blocks, regardless if they're meant for dragging or not?

Would there be some more generic way to also allow blocks to implement drag-and-dropping of non-image elements?

File: packages/block-library/src/gallery/style.scss [hunk 2]
```diff
@@ -30,6 +30,11 @@
 			display: block;
 			max-width: 100%;
 			height: auto;
+			cursor: grab;
+
+			&:active {
+				cursor: grabbing;
+			}
```
Comment (simison) [inline packages/block-library/src/gallery/style.scss]: Should these styles be applied from `Draggable` component automatically?

**Comment by noisysocks** [comment]
@nosolosw:

**Comment by noisysocks** [comment]
@nosolosw: What's the status on this one?

**Comment by nosolosw** [comment]
hey @noisysocks, sorry for the delayed response (was AFK). This is in a holding pattern at the moment. It needs more work at the design level and mechanics, but I have other priorities to address first, so not sure when I can come back to this.

**Comment by OffCorner** [comment]
would love to see this feature, was a bit surprised it wasn't there!

**Comment by noisysocks** [comment]
I'm going to close this one out as it's stalled. We can always re-open if someone decides to pick it up.

As a side note, I'd encourage us to think about refactoring the Gallery block to use `InnerBlocks` so that there's fewer design patterns for moving things around within the editor. We would then get drag and drop for free 🙂

**Pull request closed without merging**